### PR TITLE
Databricks - New Rules added

### PR DESCRIPTION
**Pull request by batemansogq**
## PR Summary

- 2 ADB rules added, min SKU and Public Access
- 
- minor text update

## PR Checklist

- [ ] PR has a meaningful title
- [ ] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule.Rules.Azure/rules/Azure.Databricks.Rule.yaml [hunk 1]
```diff
@@ -25,3 +25,43 @@ spec:
...
+    release: GA
+    ruleSet: 2024_01
+    Azure.WAF/pillar: Performance Efficiency
+spec:
+  type:
+  - Microsoft.Databricks/workspaces
+  condition:
+    allOf:
+      - field: sku.name
+        notEquals: trial
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Databricks.Rule.yaml:46]: Don't have to change it. More just a call out that this is a simpler alternative:

```suggestion
  condition:
     field: sku.name
     notEquals: trial
```

File: src/PSRule.Rules.Azure/rules/Azure.Databricks.Rule.yaml [hunk 2]
```diff
@@ -25,3 +25,43 @@ spec:
...
+    Azure.WAF/pillar: Security
+spec:
+  type:
+  - Microsoft.Databricks/workspaces
+  condition:
+    allOf:
+      - field: properties.publicNetworkAccess
+        exists: true
+      - field: properties.publicNetworkAccess
+        equals: Disabled
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Databricks.Rule.yaml:67]: Don't have to change it. More just a call out that this is a simpler alternative:

```suggestion
  condition:
    field: properties.publicNetworkAccess
    equals: Disabled
```

Referencing a field with a specific value implies the property exists. PSRule first checks the property exists, then check that the value matches the expected value.

File: docs/en/rules/Azure.Databricks.PublicAccess.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+For example:
+
+```bicep
+resource databricks 'Microsoft.Databricks/workspaces@2023-02-01' = {
+  name: name
+  location: location
+  properties: {
+    managedResourceGroupId: managedRg.id
+    publicNetworkAccess: 'Disabled'
+    }
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.PublicAccess.md:62]: Remove extra `}`.

```suggestion
```

File: docs/en/rules/Azure.Databricks.PublicAccess.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+```
+
+## LINKS
+
+- [Azure Databricks WorkspaceProperties](https://learn.microsoft.com/azure/templates/Microsoft.Databricks/workspaces?pivots=deployment-language-bicep#:~:text=WorkspaceCustomParameters-,publicNetworkAccess,-The%20network%20access)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.PublicAccess.md:69]: Add WAF reference.

```suggestion
- [SE:06 Network controls](https://learn.microsoft.com/azure/well-architected/security/networking)
- [Azure Databricks WorkspaceProperties](https://learn.microsoft.com/azure/templates/Microsoft.Databricks/workspaces?pivots=deployment-language-bicep#:~:text=WorkspaceCustomParameters-,publicNetworkAccess,-The%20network%20access)
```

File: docs/en/rules/Azure.Databricks.PublicAccess.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+reviewed: 2024-01-12
+severity: Critical
+pillar: Security
+category: Application endpoints
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.PublicAccess.md:5]: New Well-Architected Framework v2 categories are aligned to the checklist.

```suggestion
category: SE:06 Network controls
```

File: docs/en/rules/Azure.Databricks.SKU.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
+---
+reviewed: 2024-01-12
+severity: Critical
+pillar: Performance Efficiency
+category: Scaling and Partitioning
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.SKU.md:5]: [PE:03 Selecting services](https://learn.microsoft.com/azure/well-architected/performance-efficiency/select-services) might be a better fit for this one. But if you disagree, let me know.

```suggestion
category: PE:03 Selecting services
```

File: docs/en/rules/Azure.Databricks.SKU.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  location: location
+  sku: {
+    name: standard
+  }
+}
+```
+
+## LINKS
+
+- [Databricks Setup](https://learn.microsoft.com/azure/databricks/getting-started/#:~:text=Bicep-,Note,-When%20you%20create)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.SKU.md:69]: WAF link:

```suggestion
[PE:03 Selecting services](https://learn.microsoft.com/azure/well-architected/performance-efficiency/select-services)
- [Databricks Setup](https://learn.microsoft.com/azure/databricks/getting-started/#:~:text=Bicep-,Note,-When%20you%20create)
```

File: docs/en/rules/Azure.Databricks.SKU.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+## DESCRIPTION
+
+An Azure Databricks workspace has three available SKU types to support the compute demands of a workspace.
+
+The Trial SKU is a time-bound offer which has feature and compute limitations, making it unsuitable for production workloads.
+*NB* - The Trial SKU is a strong candidate for non-production or innovation workloads which can accept the tiers constraints. 
+
+## RECOMMENDATION
+
+Consider configuring Databricks workspaces to use either Standard or Premium tiers, dependant on the workload demands and NFRs.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.SKU.md:25]:
```suggestion
Consider configuring Databricks workspaces to use either Standard or Premium tiers, dependant on the workload demands and non-functional requirements (NFRs).
```

File: docs/en/rules/Azure.Databricks.SKU.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+```json
+{
+  "type": "Microsoft.Databricks/workspaces",
+  "apiVersion": "2023-02-01",
+  "name": "[parameters('name')]",
+  "location": "[parameters('location')]",
+  "sku": {
+    "name": "standard"
+  },
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Databricks.SKU.md:45]: Extra comma.

```suggestion
  }
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @batemansogq. Looks good. A few suggested changes below but otherwise all good.

**Comment by batemansogq** [comment]
comments addressed, changes applied. Please review.

**Review by BernieWhite — APPROVED** [review]
Thanks @batemansogq. Changes look great. All good to merge.

**Pull request merged**